core/trino-main/src/main/java/io/trino/operator/TopNOperator.java

```diff
@@ -136,13 +145,46 @@ public void close()
         @Override
         public Factory duplicate()
         {
-            return new Factory(operatorId, planNodeId, sourceTypes, n, sortChannels, sortOrders, typeOperators);
+            return new Factory(operatorId, planNodeId, sourceTypes, n, sortChannels, sortOrders, typeOperators, maxPartialMemory);
+        }
+    }
+
+    private final class TopNTransformingProcess
```

Comment:
We don't need `TopNTransformingProcess`. Just modifying https://github.com/trinodb/trino/pull/10166/files#r791829469 as in should be sufficient

Reply:
This is the benchmark result.

AdaptiveWorkProcessorOperator
Benchmark                   (positionsPerPage)  (topN)   Mode  Cnt   Score   Error  Units
BenchmarkTopNOperator.topN                  32       1  thrpt   60  16.917 ± 1.182  ops/s
BenchmarkTopNOperator.topN                  32     100  thrpt   60  17.369 ± 0.517  ops/s
BenchmarkTopNOperator.topN                  32   10000  thrpt   60   5.286 ± 0.058  ops/s
BenchmarkTopNOperator.topN                1024       1  thrpt   60  23.479 ± 0.387  ops/s
BenchmarkTopNOperator.topN                1024     100  thrpt   60  22.266 ± 0.267  ops/s
BenchmarkTopNOperator.topN                1024   10000  thrpt   60   7.306 ± 0.087  ops/s

WorkProcessorOperator
Benchmark                   (positionsPerPage)  (topN)   Mode  Cnt   Score   Error  Units
BenchmarkTopNOperator.topN                  32       1  thrpt   60  18.258 ± 0.512  ops/s
BenchmarkTopNOperator.topN                  32     100  thrpt   60  17.205 ± 0.331  ops/s
BenchmarkTopNOperator.topN                  32   10000  thrpt   60   5.377 ± 0.082  ops/s
BenchmarkTopNOperator.topN                1024       1  thrpt   60  22.251 ± 1.348  ops/s
BenchmarkTopNOperator.topN                1024     100  thrpt   60  21.964 ± 0.295  ops/s
BenchmarkTopNOperator.topN                1024   10000  thrpt   60   7.349 ± 0.118  ops/s

WorkProcessorOperator showed a little bit of performance degradation when positionsPerPage is bigger. Do you think we can disregard this?

Reply:
https://github.com/trinodb/trino/pull/10832

Reply:
Looking at the results, they seem like they are in the margin error.  Even if you ignore the error... my eyes say this it a few % difference, so my gut says, do the simplest thing.

Reply:
I already submit a separate PR for this. #10832 I just didn't update comments here.